src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestWarning.kt

```diff
@@ -0,0 +1,203 @@
+        "§aYou equipped a §r§.(?<reforge>\\S+)? (?<item>.*)§r§a!"
+    )
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        val message = event.message
+        equipmentPattern.matchMatcher(message) {
+            group("reforge")?.let {
+                if (it.contains("Squeaky")) {
+                    equipmentPestCooldown += 1
+                }
+            }
+            group("item")?.let {
+                equipmentPestCooldown = when (it) {
+                    "Pesthunter's Gloves" -> equipmentPestCooldown + 1
+                    "Pest Vest" -> equipmentPestCooldown + 20
+                    else -> equipmentPestCooldown
+                }
+            }
+        }
+    }
+
+    @HandleEvent
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+
+        if (event.inventoryName == "Your Equipment and Stats") {
+            equipmentPestCooldown = checkEquipment(event.inventoryItems)
+        }
+
+        WardrobeAPI.inventoryPattern.matchMatcher(event.inventoryName) {
+            if (!warningShown) return
+            wardrobeOpened = true
+        }
+    }
+
+    @HandleEvent
+    fun onSecondPassed(event: SecondPassedEvent) {
+        sprayMultiplier = checkSpray()
+        cooldown = BASE_PEST_COOLDOWN * (sprayMultiplier * (1 - equipmentPestCooldown.div(100.0)) * repellentMultiplier)
+    }
+
+    @HandleEvent(onlyOnIsland = IslandType.GARDEN)
+    fun onPestSpawn(event: PestSpawnEvent) {
+        warningShown = false
+        wardrobeOpened = false
+    }
+
+    private fun checkEquipment(equippedItems: Map<Int, ItemStack>): Int {
+        val slotsToCheck = setOf(10, 19, 28, 37)
+        var totalReduction = 0
+
+        for ((slot, itemStack) in equippedItems) {
+            if (slot !in slotsToCheck) continue
+
+            val internalName = itemStack.getInternalName().asString()
+            val reforgeName = itemStack.getReforgeName()
+
+            if (internalName == "PESTHUNTERS_GLOVES") totalReduction += 1
+            if (internalName == "PEST_VEST") totalReduction += 20
+            if (reforgeName == "squeaky") totalReduction += 1
+        }
+        return totalReduction
+    }
+
+    private fun checkSpray(): Double {
+        val plot = GardenPlotAPI.getCurrentPlot() ?: return 1.0
+        return if (plot.currentSpray == null) 1.0 else if (Perk.PEST_ERADICATOR.isActive) 0.25 else 0.5
+    }
+
+    @HandleEvent
+    fun warn(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (warningShown) return
+
+        val cooldownValue = cooldown ?: return
+        val timeSinceLastPest = PestSpawnTimer.lastSpawnTime.passedSince()
+        if (timeSinceLastPest < cooldownValue - config.pestSpawnWarningTime.seconds) return
+        LorenzUtils.sendTitle("§cPests Cooldown Expired!", duration = 3.seconds)
+
+        ChatUtils.clickableChat(
+            "§cPest spawn cooldown has expired",
```

Comment:
the message doesnt say much whats happening/what is going on/what the user should do next/why they get this message.